### ISSUE TITLE
[14.0] bi_sql_editor - filters does not works

### ISSUE THREAD

**Issue by kharkot** [OCA/reporting-engine#642]
## Module
bi_sql_editor

## Describe the bug
I have a problem with on SLQ report (it works properly in psql or external querry tools) but it I put any value in report filter sysyem does not filter data properly.
SQL is not quite  complicated:
```
select
    rc.name  as x_firma,
    rp.name as x_Kontrahent,
    sm.reference as x_NumerTransakcji,
    it.value as  x_TypTransakcji,
    spt.code as  x_RodzajTransakcji,
    sw.name as x_Magazyn,
    svl.create_date as x_DataTransakcji, 
    svl.description as x_OpisTransakcji,
    -svl.quantity as x_Ilosc, 
    round(svl.unit_cost, 4)  as x_KosztJendostkowy, 
    -svl.value as x_Wartosc,
    sm.origin as Zrodlo,
    coalesce (amd.name, amc.name) as x_DokumentKsiegowy, 
    coalesce (amd.date, amc.date) as x_DataDokumentuKsiegowego
from stock_valuation_layer svl 
    join stock_move sm on svl.stock_move_id =sm.id and (sm.location_id =5 or sm.location_dest_id =5)
    join stock_picking sp on sm.picking_id =sp.id 
    join stock_picking_type spt on spt.id =sp.picking_type_id  
    join stock_warehouse sw on sw.id=spt.warehouse_id 
    left join ir_translation it on it.res_id =spt.id and it.name='stock.picking.type,name' and it.lang ='pl_PL'
    left join account_move am on am.id =svl.account_move_id 
    left join account_move_line aml on am.id =aml.move_id and aml.account_id in (1519, 2194)
    left join account_partial_reconcile aprd on aml.id =aprd.debit_move_id 
    left join account_move_line amld on aprd.credit_move_id =amld.id 
    left join account_move amd on amd.id =amld.move_id 
    left join account_partial_reconcile aprc on aml.id =aprc.credit_move_id  
    left join account_move_line amlc on aprc.debit_move_id=amlc.id 
    left join account_move amc on amc.id =amlc.move_id 
    join res_company rc on rc.id =svl.company_id 
    join res_partner rp on rp.id =sp.partner_id
```

look at attachment :
1. in fiter i have: 'Numertransakcji is equal to "WZ-CHTR/22/00017"'
2 in result I have row with totally different 'Numertransakcji'

<img width="1273" alt="Zrzut ekranu 2022-08-23 o 23 42 43" src="https://user-images.githubusercontent.com/43400881/186271996-e3599b1c-ed02-4c3f-90eb-343f6489eb1b.png">

**Comment by AndrzejGerasimukARCHIMEDES** [comment]
I have prepare a view on database to the same SQL and made simple Select to the definition.


The similar problem is also when i group by specifier field. Is shows rows with different values.

![obraz](https://user-images.githubusercontent.com/112691008/191911548-15cf0876-834d-4891-9178-3d7838026362.png)

When i export all data to XLS first rows was OK, but most of data did not match with real results given direct form database.

Maybe this will be tip for developer:
When i add in my SQL VIEW order by "Numertransakcji" (filed sm.reference) the result on your model, on export to file, on group by or on filter is OK.

**Comment by kharkot** [comment]
Please  apply the "no stale" label

**Comment by KatiusciaGusmao** [comment]
I had the same issue with the filter view @kharkot  . 
When I tried the filter by date, some kind of filtering occurred, but it was incorrect and I couldn't filter properly.
I flowed the way of @AndrzejGerasimukARCHIMEDES , and I included "order by" in my query and worked.
So, is there a way to correct this in the code? or is it always necessary to include an "order by" for the filter to work correctly?
